src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,63 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirm extends Popup {
+  constructor(popupSelector) {
+    super(popupSelector);
+    this._form = this._popup.querySelector(".modal__form");
+    this._submitButton = this._form.querySelector(".modal__button");
+    this._defaultButtonText = this._submitButton
+      ? this._submitButton.textContent
+      : "Yes";
+  }
+
+  setAction(action) {
+    this._handleSubmit = action;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._form.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this._handleSubmit();
+    });
+  }
+
+  open() {
+    super.open();
+    if (this._submitButton) {
+      this._submitButton.textContent = this._defaultButtonText;
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.